### PR TITLE
Improve test coverage for ClinicIO

### LINKED ISSUE #242: XmlAdaptedPatient missing no-arg constructor
**Describe the bug**
XmlAdaptedPatient lacks the no-arg constructor which is required by JAXB.

The following error is caused by this:
> java.lang.NoSuchMethodException: seedu.clinicio.storage.XmlAdaptedPatient.<init>()

### PR DESCRIPTION
**Pull request by jjlee050**
Work on improving test coverage 
(Currently at 63% - 71%)
**PS: Apologies for editing/adding so much class at one go.**

Logic:
- `AddPatientCommandTest.java`
- `AddPatientCommandIntegrationTest.java`
- `AddPatientCommandParserTest.java`
- `EditCommandTest.java`
- `DoctorStatisticsCommandTest.java`
- `PatientStatisticsCommandTest.java`
- `AppointmentStatisticsCommandTest.java`
- `CancelApptCommandParserTest.java`

Model:
- `ClinicIoTest.java`
- `ModelManagerTest.java`
- `StaffTest.java`
- `PasswordTest.java`
- `UniqueStaffList.java`
- `NricTest.java`
- `MedicalProblemTest.java`
- `MedicationTest.java`
- `AllergyTest.java`

Storage:
- `XmlAdaptedStaffTest.java`: Fix #242 

UI
- `StatusBarFooterTest.java`

### CODE REVIEW

**Comment by jjlee050** [comment]
@arsalanc-v2 @gingivitiss I called u guys becoz i make changes to some of ur files. Take a look and lmk if there is any issues ASAP.

**Comment by arsalanc-v2** [comment]
What changes did you make for mine?

**Comment by jjlee050** [comment]
ps i am still working on it coz the xmladaptedPatient. preferredDoc is optional<staff> and appointment is Optional<Appointment> I think this is affecting several classes which include urs.

**Comment by jjlee050** [comment]
I will continue working on the test coverage after this PR is approve.

File: src/test/java/seedu/clinicio/logic/parser/ClinicIoParserTest.java [hunk 1]
```diff
@@ -22,18 +22,29 @@
...
+import seedu.clinicio.logic.commands.ExportPatientsAppointmentsCommand;
+import seedu.clinicio.logic.commands.ExportPatientsCommand;
+import seedu.clinicio.logic.commands.ExportPatientsConsultationsCommand;
 import seedu.clinicio.logic.commands.FindCommand;
 import seedu.clinicio.logic.commands.HelpCommand;
 import seedu.clinicio.logic.commands.HistoryCommand;
 import seedu.clinicio.logic.commands.ListCommand;
 import seedu.clinicio.logic.commands.LoginCommand;
+import seedu.clinicio.logic.commands.LogoutCommand;
+import seedu.clinicio.logic.commands.MedicineStatisticsCommand;
```
Comment (arsalanc-v2) [inline src/test/java/seedu/clinicio/logic/parser/ClinicIoParserTest.java]:
```suggestion
```

File: src/test/java/seedu/clinicio/logic/parser/ClinicIoParserTest.java [hunk 2]
```diff
@@ -129,7 +170,39 @@ public void parseCommand_list() throws Exception {
...
+    @Test
+    public void parseCommand_doctorStatistics() throws Exception {
+        assertTrue(parser
+                .parseCommand(DoctorStatisticsCommand.COMMAND_WORD) instanceof DoctorStatisticsCommand);
+        assertTrue(parser
+                .parseCommand(DoctorStatisticsCommand.COMMAND_WORD + " 3") instanceof DoctorStatisticsCommand);
+    }
+
+    @Test
+    public void parseCommand_medicineStatistics() throws Exception {
```
Comment (arsalanc-v2) [inline src/test/java/seedu/clinicio/logic/parser/ClinicIoParserTest.java]: Won't be working on medicine stats, you can remove this.

Comment (jjlee050) [reply src/test/java/seedu/clinicio/logic/parser/ClinicIoParserTest.java]: I will remove the class so that the test coverage wont consider it.

**Pull request merged**